javalib/src/main/scala/java/util/AbstractList.scala

```diff
@@ -109,63 +116,59 @@ abstract class AbstractList[E] protected () extends AbstractCollection[E]
     iter.take(toIndex - fromIndex).foreach(_ => iter.remove())
   }
 
-  protected[this] def checkIndexInBounds(index: Int): Unit = {
+  protected def checkIndexInBounds(index: Int): Unit = {
     if (index < 0 || index >= size)
       throw new IndexOutOfBoundsException(index.toString)
   }
 
-  protected[this] def checkIndexOnBounds(index: Int): Unit = {
+  protected def checkIndexOnBounds(index: Int): Unit = {
     if (index < 0 || index > size)
       throw new IndexOutOfBoundsException(index.toString)
   }
+}
 
-  private abstract class AbstractListView(fromIndex: Int,
-      private var toIndex: Int) extends AbstractList[E] {
-
-    override def add(index: Int, e: E): Unit = {
-      checkIndexOnBounds(index)
-      self.add(fromIndex + index, e)
-      toIndex += 1
-    }
-
-    override def addAll(index: Int, c: Collection[_ <: E]): Boolean = {
-      checkIndexOnBounds(index)
-      self.addAll(fromIndex + index, c)
-      val elementsAdded = c.size
-      toIndex += elementsAdded
-      elementsAdded != 0
-    }
+private abstract class AbstractListView[E](protected val list: List[E],
+    fromIndex: Int, private var toIndex: Int) extends AbstractList[E] {
 
-    override def addAll(c: Collection[_ <: E]): Boolean =
-      addAll(size, c)
+  override def add(index: Int, e: E): Unit = {
+    checkIndexOnBounds(index)
+    list.add(fromIndex + index, e)
+    toIndex += 1
```

Comment:
Maybe this, and other updates to `toIndex` in this class, should now call `changeViewSize` instead?